src/Elastic.Apm.AspNetCore/ApmMiddleware.cs

```diff
@@ -42,21 +42,38 @@ public ApmMiddleware(RequestDelegate next, Tracer tracer, IApmAgent agent)
 
 		public async Task InvokeAsync(HttpContext context)
 		{
-			var transaction = StartTransaction(context);
+			var transaction = await StartTransactionAsync(context);
 
 			try
 			{
+				// We need to reset the current transaction (which is async local), since the StartTransactionAsync call above ended with all its
+				// child spans at this point, therefore need to set the current transaction here again.
+				_tracer.CurrentExecutionSegmentsContainer.CurrentTransaction = transaction;

```

Comment:
IMHO this breaks encapsulation - a possible alternative would be to extract `await FillSampledTransactionContextRequest` from `StartTransactionAsync` which will revert `StartTransaction` to being sync.

Reply:
No string feelings on this either. I suspect we'll run into this more later.

Changed this so we can move on.